src/jit/emitxarch.cpp

```diff
@@ -214,7 +217,8 @@ bool TakesRexWPrefix(instruction ins, emitAttr attr)
     // cased here.
     //
     // Rex_jmp = jmp with rex prefix always requires rex.w prefix.
-    if (ins == INS_movsx || ins == INS_rex_jmp)
+    // vpermq requires w bit to set to 1
+    if (ins == INS_movsx || ins == INS_rex_jmp || ins == INS_vpermq)
     {
```

Comment:
We already have `TakesVexPrefix()` and `AddVexPrefix()`. Maybe what you want is a function that returns true if the VEX prefix W bit needs to be set (I assume nobody required this until now). Perhaps that should be called `RequiresVexPrefixWBit(instruction ins)`? Then, `AddVexPrefix()` could have code like:
```
if (RequiresVexPrefixWBit(ins))
{
   code |= ... <magic W bit mask> ...
}
```